### PR TITLE
Added idle bars

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
Adds Idle Bars, that behave like progress bars, but do not show a linear progress and instead only indicate that something is happening. They can be used, when the amount of operations is unknown.

### CODE REVIEW

**Comment by Kr0nox** [comment]
Works good in windows terminals. 
Only thing I noticed was that the bouncer did not reach the left end full. There alsways was a space between `<` and `<+>`.

**Comment by Kr0nox** [comment]
I could not see it hit the left side a single time. So maybe there actually is a windows issue here?

**Comment by Kr0nox** [comment]
yes, with this delay it works

**Review by tsaglam — COMMENTED** [review]
Tested the main method under macOS, both in IDE and in the terminal. Looks good, no issues. Adapts to terminal window size.

**Comment by Kr0nox** [comment]
@TwoOfTwelve I dont get a progess bar during clustering is that on purpose?

File: core/src/main/java/de/jplag/clustering/ClusteringFactory.java [hunk 1]
```diff
@@ -61,6 +66,8 @@ public static List<ClusteringResult<Submission>> getClusterings(Collection<JPlag
 
         // remove bad clusters
         result = removeBadClusters(result);
+
+        progressBar.dispose();
         logClusters(result);
```
Comment (Kr0nox) [inline core/src/main/java/de/jplag/clustering/ClusteringFactory.java:71]: I would not printing all the clusters to the console

**Review by tsaglam — APPROVED** [review]

**Pull request merged**